### PR TITLE
Project 17 Express API

### PR DESCRIPTION
**Pull request by Ann-Sofi**
Links to the live project: 

Start: https://books-api-2021.herokuapp.com/

Example of endpoints:  

https://books-api-2021.herokuapp.com/top5

https://books-api-2021.herokuapp.com/textreviews

https://books-api-2021.herokuapp.com/isbn/1400052920

https://books-api-2021.herokuapp.com/books?author=J.K.%20Rowling&title=prisoner&language=eng

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,34 +1,116 @@
...
+
+  if (title) {
+    bookList = bookList
+      .filter((book) => book.title.toString().toLowerCase().includes(title.toLowerCase()))
+  }
+
+  if (language) {
+    bookList = bookList
+      .filter((book) => book.language_code.toLowerCase().includes(language.toLowerCase()))
+  }
```
Comment (KarolCislowski) [inline server.js:47]: nice filtering

File: server.js [hunk 2]
```diff
@@ -1,34 +1,116 @@
...
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be
```
Comment (KarolCislowski) [inline server.js:14]: nice cleaning 🧹

File: server.js [hunk 3]
```diff
@@ -1,34 +1,116 @@
...
+  if (bookList.length) {
+    res.json(bookList)
+  } else {
+    res.status(404).send("No match");
+  }
+});
+
+// endpoint to get the top 5 books sorted on rating
+app.get('/top5', (req, res) => {
+  const books = booksData.sort((a, b) => b.average_rating - a.average_rating);
```
Comment (KarolCislowski) [inline server.js:58]: you used sorting a few times. maybe it is a good idea to create a sorting function?

File: server.js [hunk 4]
```diff
@@ -1,34 +1,116 @@
...
+});
+
+// endpoint to get the 20 books with most textreviews
+app.get('/textreviews', (req, res) => {
+  const books = booksData.sort((a, b) => b.text_reviews_count - a.text_reviews_count);
+  res.json(books.slice(0, 20));
 })
 
+// endpoint for isbn
+app.get('/isbn/:isbn', (req, res) => {
```
Comment (KarolCislowski) [inline server.js:69]: you are filtering books, so proper address should be `/books/isbn/:isbn`

File: server.js [hunk 5]
```diff
@@ -1,34 +1,116 @@
...
+  const books = booksData.find((book) => book.bookID === +id)
+  if (!books) {
+    res.status(404).send("Book id not found");
+  } else {
+    res.json(books)
+  }
+});
+
+// endpoint to filter on language
+app.get('/lang/:lang', (req, res) => {
```
Comment (KarolCislowski) [inline server.js:91]: `/books/lang/:lang` 🤓

File: server.js [hunk 6]
```diff
@@ -1,34 +1,116 @@
...
+  const books = booksData.find((book) => book.isbn === +isbn)
+  if (!books) {
+    res.status(404).send("Book isbn not found");
+  } else {
+    res.json(books)
+  }
+});
+
+// endpoint for id
+app.get('/id/:id', (req, res) => {
```
Comment (KarolCislowski) [inline server.js:80]: `/books/:id` 

not `/books/id/:id` because the id is a primary key 🤓

**Review by KarolCislowski — APPROVED** [review]
Nice API 🌈

File: server.js (hunk 6 above)
Comment (Ann-Sofi) [reply server.js:80]: Could you please explain what a  primary key is? :-)

**Pull request closed without merging**